LogicalTypes.md

```diff
@@ -767,6 +767,188 @@ optional group my_map (MAP_KEY_VALUE) {
+A geometry has at least two coordinate dimensions: X and Y for 2D coordinates
+of each point. A geometry can optionally have Z and / or M values associated
+with each point in the geometry.
+
+The Z values introduce the third dimension coordinate. Usually they are used
+to indicate the height, or elevation.
+
+M values are an opportunity for a geometry to express a fourth dimension as
+a coordinate value. These values can be used as a linear reference value
+(e.g., highway milepost value), a timestamp, or some other value as defined
+by the CRS.
+
+Bounding box is defined as the thrift struct below in the representation of
+min/max value pair of coordinates from each axis. Note that X and Y Values
+are always present. Z and M are omitted for 2D geometries.
+
+```thrift
+struct BoundingBox {
+  /** Min X value when edges = PLANAR, westmost value if edges = SPHERICAL */
+  1: required double xmin;
+  /** Max X value when edges = PLANAR, eastmost value if edges = SPHERICAL */
+  2: required double xmax;
+  /** Min Y value when edges = PLANAR, southmost value if edges = SPHERICAL */
+  3: required double ymin;
+  /** Max Y value when edges = PLANAR, northmost value if edges = SPHERICAL */
+  4: required double ymax;
+  /** Min Z value if the axis exists */
+  5: optional double zmin;
+  /** Max Z value if the axis exists */
+  6: optional double zmax;
+  /** Min M value if the axis exists */
+  7: optional double mmin;
+  /** Max M value if the axis exists */
+  8: optional double mmax;
+}
+```
+
+The meaning of each value depends on the `Edges` attribute of the `GEOMETRY` type:
+- If Edges is `PLANAR`, the values are literally the actual min/max value from each axis.
+- If Edges is `SPHERICAL`, the values for X and Y are `[westmost, eastmost, southmost, northmost]`,
+  with necessary min/max values for Z and M if needed.
+
+##### Geometry Types
+
+A list of geometry types from all geometries in the `GEOMETRY` column, or an
+empty list if they are not known.
+
+This is borrowed from [geometry_types of GeoParquet][geometry-types]
+except that values in the list are [WKB (ISO-variant) integer codes][wkb-integer-code].
+Table below shows the most common geometry types and their codes:
+
+| Type               | XY   | XYZ  | XYM  | XYZM |
+| :----------------- | :--- | :--- | :--- | :--: |
+| Point              | 0001 | 1001 | 2001 | 3001 |
+| LineString         | 0002 | 1002 | 2002 | 3002 |
+| Polygon            | 0003 | 1003 | 2003 | 3003 |
+| MultiPoint         | 0004 | 1004 | 2004 | 3004 |
+| MultiLineString    | 0005 | 1005 | 2005 | 3005 |
+| MultiPolygon       | 0006 | 1006 | 2006 | 3006 |
+| GeometryCollection | 0007 | 1007 | 2007 | 3007 |
+
+In addition, the following rules are applied:
+- A list of multiple values indicates that multiple geometry types are present (e.g. `[0003, 0006]`).
+- An empty array explicitly signals that the geometry types are not known.
+- The geometry types in the list must be unique (e.g. `[0001, 0001]` is not valid).
```

Comment:
What's the purpose of this list if it is so unspecified? Any conforming writer implementation can choose to never populate it.

Reply:
IIUC, it should be empty if we have any invalid geometry data in the column. `ST_IsValid` deals with this case: https://postgis.net/docs/ST_IsValid.html.